tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Http/WebRequest/HttpWebRequest_EndGetResponse_Integration.cs

```diff
@@ -75,14 +76,17 @@ public class HttpWebRequest_EndGetResponse_Integration
 
                     if (scope is not null)
                     {
-                        if (setSamplingPriority && existingSpanContext?.SamplingPriority is not null)
+                        var span = scope.Span;
+
+                        if (setSamplingPriority)
                         {
-                            scope.Span.SetTraceSamplingPriority(existingSpanContext.SamplingPriority.Value);
+                            // TODO: figure out SamplingMechanism, do we propagate that as well for special cases like this?
```

Comment:
TODO: figure out SamplingMechanism, do we propagate that as well for special cases like this?